docs/release-notes/index.md

```diff
@@ -1,5 +1,5 @@
 ---
-navigation_title: "Release notes"
+navigation_title: "Elastic APM Android Agent"
```

Comment:
We need to update this (at least for now) because of how this title is used when the docs are compiled. Right now there's no indication that the top section in the image below is related to the OTel Android Agent. (I went with `Elastic APM Android Agent` for consistency, but let me know if we should change it.)

<img width="296" alt="image" src="https://github.com/user-attachments/assets/a4b3dbb1-9401-48a8-9cab-de50e722ae4f" />

Reply:
Got it, thanks!